### ISSUE TITLE
Build policy 'always' fails on Windows

### ISSUE THREAD

**Issue by theodelrieu** [conan-io/conan#1956]
Hi there,

I'm currently trying to add Conan support to [Outcome](https://github.com/ned14/Outcome/pull/68).
For that purpose I chose to use `build_policy = "always"`.

It works on Linux, but when I tried to create a package on my Windows machine, I got the following output:

```
Outcome/master@theo/test: Exporting package recipe
WARN: Conanfile doesn't have 'url'.
It is recommended to add it as attribute
Outcome/master@theo/test: A new conanfile.py version was exported
Outcome/master@theo/test: Folder: C:\Users\Theo\.conan\data\Outcome\master\theo\test\export
Outcome/master@theo/test: Package recipe modified in export, forcing source folder removal
Outcome/master@theo/test: Use the --keep-source, -k option to skip it
Outcome/master@theo/test: Removing 'source' folder, this can take a while for big packages
Outcome/master@theo/test: Installing package
Requirements
    Outcome/master@theo/test from godzilla
Packages
    Outcome/master@theo/test:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9

Outcome/master@theo/test: Building package from source as defined by build_policy='always'
Outcome/master@theo/test: Building package from source as defined by build_policy='always'
Outcome/master@theo/test: Building package from source as defined by build_policy='always'
Outcome/master@theo/test: WARN: Forced build from source
Outcome/master@theo/test: Building your package in C:\Users\Theo\.conan\data\Outcome\master\theo\test\build\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
Outcome/master@theo/test: WARN: Detected build_policy 'always', trying to remove source folder
Outcome/master@theo/test: WARN: This can take a while for big packages
Outcome/master@theo/test: Configuring sources in C:\Users\Theo\.conan\data\Outcome\master\theo\test\source


Outcome/master@theo/test: Copying sources to build folder
Outcome/master@theo/test: Generator txt created conanbuildinfo.txt
Outcome/master@theo/test: Calling build()
Outcome/master@theo/test: WARN: This conanfile has no build step
Outcome/master@theo/test: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' built
Outcome/master@theo/test: Build folder C:\Users\Theo\.conan\data\Outcome\master\theo\test\build\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
Outcome/master@theo/test: Generated conaninfo.txt
Outcome/master@theo/test: Generated conanbuildinfo.txt
Outcome/master@theo/test: Generating the package
Outcome/master@theo/test: Package folder C:\Users\Theo\.conan\data\Outcome\master\theo\test\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
Outcome/master@theo/test: Calling package()
Outcome/master@theo/test package(): Copied 1 '.hpp' files: outcome.hpp
Outcome/master@theo/test package(): Copied 1 '' files: LICENCE
Outcome/master@theo/test: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' created
Outcome/master@theo/test: Testing with 'test_package'
Outcome/master@theo/test test package: Installing dependencies
Outcome/master@theo/test: Building package from source as defined by build_policy='always'
Outcome/master@theo/test: Building package from source as defined by build_policy='always'
Outcome/master@theo/test: Building package from source as defined by build_policy='always'
Outcome/master@theo/test: WARN: Forced build from source
ERROR: [WinError 32] Le processus ne peut pas acc▒der au fichier car ce fichier est utilis▒ par un autre processus: 'C:\\Users\\Theo\\.conan\\data\\Outcome\\master\\theo\\test\\build\\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9'

Couldn't remove folder, might be busy or open
Close any app using it, and retry
```

The Windows error means: a process is already using this file. Pardon my french :)

I've tried with versions 0.28 and 0.27, it fails too. As you can see some outputs are repeated three times, it did the same thing on Linux.

Note that the package is header only, nothing is to be built, but Conan still seems to create a build folder, even if there is no `build()` method in the `conanfile.py`. From the Windows error I assume that some task is trying to remove the build folder while it's being used.

The recipe can be found [here](https://github.com/ned14/outcome/tree/develop/conan)

**Comment by memsharded** [comment]
I am having a look. My main issue is that the URL for downloading the files is wrong (or github raw content is not working). I have tried with and without ``build_policy``. Can you please check it?

Trying with another URL it worked here (though it failed building the test_package example.cpp)

French is good. J'etudie an peu a l'ecole, I can read it :)

**Comment by memsharded** [comment]
It seems there could be an error with ``tools.download`` and github raw content, I am getting the following error:
```
ERROR: Outcome/master@user/testing: Error in source() method, line 13
        tools.download("https://raw.githubusercontent.com/ned14/outcome/master/single-header/outcome.hpp", filename="outcome.hpp")
        ConanConnectionError: Download failed, check server, possibly try again
Transfer interrupted before complete: 271503 < 34432
```

**Comment by theodelrieu** [comment]
I'll try without the build_policy, I also had this error earlier.

**Comment by theodelrieu** [comment]
I had no issues without the `build_policy="always"`, couldn't trigger the `Transfer interrupted` bug again...

**Comment by memsharded** [comment]
So weird, I am still seeing the ``Transfer interrupted...``, don't know what is happening.

**Comment by theodelrieu** [comment]
Strange indeed... I won't be available for the next few days, I'll try to investigate once I return.

Meanwhile I hope you can figure it out! No pressure ;)

**Comment by memsharded** [comment]
It seems there is a bug in the download when ``content-encoding=gzip`` and ``content-length`` showing a different length (the compressed one). This might deserve a minor release, created branch: https://github.com/conan-io/conan/tree/release/0.28.1 to check CI, will consider it.

**Comment by memsharded** [comment]
Now, the download is working fine, but I am able to reproduce this. It indeeds seems a bug. Investigating.

**Comment by memsharded** [comment]
It was a bug. Submitted a fix in #1974, so it will be included in next 0.29. Please avoid the ``build_policy="always"`` at the moment, if you want to get the same security, make sure to ``conan remove <yourpkg> -f`` before doing ``conan create``

Thanks again very much for all your help.

**Comment by memsharded** [comment]
Released in 0.29
